agent/auto-config/auto_encrypt_test.go

```diff
@@ -157,9 +157,20 @@ func TestAutoEncrypt_hosts(t *testing.T) {
 		"split-host-port-error": {
 			serverProvider: providerNone,
 			config: &config.RuntimeConfig{
-				RetryJoinLAN: []string{"this-is-not:a:ip:and_port"},
+				RetryJoinLAN: []string{"[this-is-not]:a:ip:and_port"},
 			},
 			err: "no auto-encrypt server addresses available for use",
+			// Something that looks like IPv6 [addr]:port syntax, but with too many colons, does trigger an error at
+			// this layer.
+		},
```

Comment:
This test no longer passed as is, as it was testing a specific failure mode of the address handling logic - so I updated this case's test data to something which would fail in the same way...